File: CS/WorkbookProgressSample/Form1.cs
using System;
using System.Threading;
using System.Windows.Forms;
using DevExpress.Spreadsheet;

namespace WorkbookProgressSample {
    public partial class Form1 : Form {
        CancellationTokenSource cancellationSource;

        public Form1() {
            InitializeComponent();
        }

        private async void RunCancel_Click(object sender, EventArgs e) {
            if (cancellationSource != null) {
                cancellationSource.Cancel();
            }
            else {
                progressBarLoad.Value = 0;
                progressBarExport.Value = 0;
                btnRunCancel.Text = "Cancel";
                cancellationSource = new CancellationTokenSource();
                try {
                    using (Workbook workbook = new Workbook()) {
                        await workbook.LoadDocumentAsync("Document.xlsx",
                            cancellationSource.Token,
                            new Progress<int>((progress) => {
                                progressBarLoad.Value = progress;
                                progressBarLoad.Refresh();
                            }));
                        await workbook.ExportToPdfAsync("Result.pdf",
                            cancellationSource.Token,
                            new Progress<int>((progress) => {
                                progressBarExport.Value = progress;
                                progressBarExport.Refresh();
                            }));
                    }
                }
                catch (OperationCanceledException) {
                    progressBarLoad.Value = 0;
                    progressBarExport.Value = 0;
                }
                finally {
                    cancellationSource.Dispose();
                    cancellationSource = null;
                    btnRunCancel.Text = "Run";
                }
            }
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e) {
            cancellationSource?.Cancel();
        }
    }
}

File: VB/WorkbookProgressSample/Form1.vb
Imports System
Imports System.Threading
Imports System.Windows.Forms
Imports DevExpress.Spreadsheet

Namespace WorkbookProgressSample

    Public Partial Class Form1
        Inherits Form

        Private cancellationSource As CancellationTokenSource

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Async Sub RunCancel_Click(ByVal sender As Object, ByVal e As EventArgs)
            If cancellationSource IsNot Nothing Then
                cancellationSource.Cancel()
            Else
                progressBarLoad.Value = 0
                progressBarExport.Value = 0
                btnRunCancel.Text = "Cancel"
                cancellationSource = New CancellationTokenSource()
                Try
                    Using workbook As Workbook = New Workbook()
                        Await workbook.LoadDocumentAsync("Document.xlsx", cancellationSource.Token, New Progress(Of Integer)(Sub(progress)
                            progressBarLoad.Value = progress
                            progressBarLoad.Refresh()
                        End Sub))
                        Await workbook.ExportToPdfAsync("Result.pdf", cancellationSource.Token, New Progress(Of Integer)(Sub(progress)
                            progressBarExport.Value = progress
                            progressBarExport.Refresh()
                        End Sub))
                    End Using
                Catch __unusedOperationCanceledException1__ As OperationCanceledException
                    progressBarLoad.Value = 0
                    progressBarExport.Value = 0
                Finally
                    cancellationSource.Dispose()
                    cancellationSource = Nothing
                    btnRunCancel.Text = "Run"
                End Try
            End If
        End Sub

        Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs)
            cancellationSource?.Cancel()
        End Sub
    End Class
End Namespace
